source/statements.tex

```diff
@@ -1038,11 +1038,28 @@
 the first time control passes through its declaration; such a variable is
 considered initialized upon the completion of its initialization. If the
 initialization exits by throwing an exception, the initialization is not
-complete, so it will be tried again the next time control enters the
+complete, so it will be tried again the next time control passes through the
 declaration.
-If control enters the declaration concurrently while the variable is
+If control passes through the declaration concurrently while the variable is
 being initialized, the concurrent execution shall wait for completion
 of the initialization.
+\begin{example}
+\begin{codeblock}
+void f() {
+  static int x = 0;                     // OK
+  static std::string s = "hello";       // OK; if an exception is thrown during the
+                                        // initialization of \tcode{s}, \tcode{s} remains uninitialized, and
+                                        // an attempt to initialize \tcode{s} (but not \tcode{x}) will be
+                                        // made the next time \tcode{f} is called
+  thread_local int n = 0;               // OK, initializes \tcode{n} to \tcode{0} on the main thread
```

Comment:
`n` (of both threads) is constant-initialized here. So the current example is seemly free from UB and may be misleading, since `n` is unrelated to dynamic initialization.